lib/ui/text.dart

```diff
@@ -947,6 +947,68 @@ class ParagraphConstraints {
   String toString() => '$runtimeType(width: $width)';
 }
 
+/// Defines various ways to vertically bound the boxes returned by
+/// [Paragraph.getBoxesForRange].
+enum BoxHeightStyle {
+    /// Provide tight bounding boxes that fit heights per run. This style may result
+    /// in uneven bounding boxes that do not nicely connect with the boxes above,
+    /// below, and adjacent.
+    tight,
+
+    /// The height of the boxes will be the maximum height of all runs in the
+    /// line. All boxes in the same line will be the same height. This does not
+    /// guarantee that the boxes will cover the entire vertical height of the line
+    /// when there is additional line spacing.
+    ///
+    /// See [RectHeightStyle.includeLineSpacingTop], [RectHeightStyle.includeLineSpacingMiddle],
+    /// and [RectHeightStyle.includeLineSpacingBottom] for styles that will cover
+    /// the entire line.
+    max,
+
+    /// Extends the top and bottom edge of the bounds to fully cover any line
+    /// spacing.
+    ///
+    /// The top and bottom of each box will cover half of the
+    /// space above and half of the space below the line. The text should be
+    /// centered vertically within the box.
```

Comment:
it might not actually be vertically centered, consider a line with a high line height containing a text span with a relatively small font size. the text will appear to be vertically near the bottom.